### PR TITLE
feat: stage unmanaged depgraph-endpoints

### PR DESCRIPTION
**Pull request by dekelund**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?

By updating to latest unmanaged-deps API version, we allow user to list package URL with print-deps.
This also requires us to update to the latest snyk-cpp-plugin, which also enables us to list the package URL in the human readable format.

We have also included a commit, to set endpoints to latest GA. This is just in case we need to rollback by doing a `git revert`.

#### How should this be manually tested?

Make sure the package URL (aka purl) is listed in the output for following two commands:

- `snyk test --unmanaged --json`
- `snyk test --unmanaged --print-deps`

#### Screenshots

Example human readable:

```
Dependencies:


  https://curl.se|curl@7.29.0
  purl: pkg:generic/curl@7.29.0?download_url=https:%2F%2Fcurl.se%2Fdownload%2Farcheology%2Fcurl-7.29.0.tar.gz
  confidence: 1.000

  https://github.com|nih-at/libzip@1.8.0
  purl: pkg:generic/libzip@1.8.0?download_url=https:%2F%2Fgithub.com%2Fnih-at%2Flibzip%2Farchive%2Fv1.8.0.tar.gz
  confidence: 1.000

  https://github.com|madler/zlib@1.2.11
  purl: pkg:generic/zlib@1.2.11?download_url=https:%2F%2Fzlib.net%2Ffossils%2Fzlib-1.2.11.tar.gz
  confidence: 1.000
```

Example json-output:

```
    "fileSignaturesDetails": {
      "https://curl.se|curl@7.29.0": {
        "artifact": "curl",
        "version": "7.29.0",
        "author": "curl",
        "path": "curl-7.29.0",
        "id": "2a0f545204ed3ffe25fa4b8c00000000",
        "url": "https://curl.se/download/archeology/curl-7.29.0.tar.gz",
        "purl": "pkg:generic/curl@7.29.0?download_url=https:%2F%2Fcurl.se%2Fdownload%2Farcheology%2Fcurl-7.29.0.tar.gz",
        "score": 1,
        "confidence": 1,
        "filePaths": [
          "curl-7.29.0/Android.mk",
          "curl-7.29.0/CHANGES",
          "curl-7.29.0/CMake/CMakeConfigurableFile.in",
          "curl-7.29.0/CMake/CurlCheckCSourceCompiles.cmake",

```

### CODE REVIEW

**Comment by dekelund** [comment]
Looks like "update branch against master/main" in github is not compliant with our rules:

```
"[Merge branch 'master' into feat/stage-unmanaged-depgraphs-endpoints](https://api.github.com/repos/snyk/cli/git/commits/bd7a0ec672957772179d327faeaf67800c3901cc)" is not using a valid commit message format. For commit guidelines, see: [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md#creating-commits).
```

File: package.json [hunk 1]
```diff
@@ -114,7 +114,7 @@
     "rimraf": "^2.6.3",
     "semver": "^6.0.0",
     "snyk-config": "4.0.0",
-    "snyk-cpp-plugin": "2.22.2",
+    "snyk-cpp-plugin": "^2.24.0",
```
Comment (darscan) [inline package.json:117]: Please pin to an explicit version - the CLI is deployed as a unit, and we need to make sure a given CLI version always uses the same plugin versions.

**Review by darscan — CHANGES_REQUESTED** [review]
`snyk-cpp-plugin` needs to be pinned at a concrete version.

File: package.json (hunk 1 above)
Comment (dekelund) [reply package.json]: My mistake, not intentional

**Comment by bastiandoetsch** [comment]
Hey @dekelund , would you please add acceptance tests for the testing laid our in the PR description?

**Review by bastiandoetsch — APPROVED** [review]

**Review by darscan — APPROVED** [review]

**Pull request merged**